src/nnet3/nnet-compute.h

```diff
@@ -124,6 +157,29 @@ class NnetComputer {
   void GetOutputDestructive(const std::string &output_name,
                             CuMatrix<BaseFloat> *output);
 
+  // Get the matrices where store the state for next computation
+  inline const std::vector< CuMatrix<BaseFloat> > &GetMatrices() const { 
+    return matrices_; }
+
+  // Get state of streams. The parameter named "batch_first" indicates whether 
+  // the matrices saved state is batch first or not.
+  void GetState(const std::vector<bool> &batch_first,
+                const int32 batch_size,
+                std::vector< NnetComputeState* > *statuses);
+  
+  // Set state of streams, The parameter named "batch_first" indicates whether 
+  // the matrices saved state is batch first or not.
+  void SetState(const std::vector<bool> &batch_first,
+                const int32 batch_size,
+                const std::vector< NnetComputeState* > &state);
+
+  // Return true if all the members are equal to other's.
+  bool Equal(const NnetComputer &other);
+
+  // Take a snapshot of the NnetComputer, discard the data saved in matrices_
+  void GetSnapshot(NnetComputerSnapshot *snapshot);
```

Comment:
Is this method const, w.r.t. the enclosing class?  If so, add "const", i.e. 
```void GetSnapshot(NnetComputerSnapshot *snapshot) const;```
I don't know what you mean by 'discard the data saved in matrices_`.  I think you should not say that;
this function does not appear to modify *this, so it doesn't affect that data.